app/src/main/java/com/github/se/eventradar/model/EventsOverviewViewModel.kt

```diff
@@ -60,27 +60,49 @@ class EventsOverviewViewModel(db: FirebaseFirestore = Firebase.firestore) : View
         }
   }
 
-  private val userRef = db.collection("users")
-
-  private fun getEventContact(contactId: String): String {
-    var contactEmail = ""
-
-    userRef
-        .document(contactId)
-        .collection("private")
-        .limit(1) // Limit the query to retrieve only one document
+  fun getUpcomingEvents(uid: String) {
+    eventRef
+        .whereArrayContains("attendees_list", uid)
```

Comment:
In theory we keep track of this for each user: https://github.com/swent-sp-2024-event-radar/event-radar-app?tab=readme-ov-file#users-database

Reply:
> In theory we keep track of this for each user: https://github.com/swent-sp-2024-event-radar/event-radar-app?tab=readme-ov-file#users-database

Great catch! 

> In theory we keep track of this for each user: https://github.com/swent-sp-2024-event-radar/event-radar-app?tab=readme-ov-file#users-database

Great catch! In my latest commit, I optimized getUpcomingEvents to utilize User's eventsAttendeeList, improving query efficiency.

Reply:
> In theory we keep track of this for each user: https://github.com/swent-sp-2024-event-radar/event-radar-app?tab=readme-ov-file#users-database

Great catch!  In my latest commit, I optimized getUpcomingEvents to utilize User's eventsAttendeeList, improving query efficiency.